esp-hal/src/interrupt/riscv.rs

```diff
@@ -452,7 +463,16 @@ mod vectored {
         unsafe {
             let ptr =
                 &pac::__EXTERNAL_INTERRUPTS[interrupt as usize]._handler as *const _ as *mut usize;
-            ptr.write_volatile(handler.raw_value());
+
+            if crate::debugger::debugger_connected() {
+                ptr.write_volatile(handler.raw_value());
+            } else {
+                LOCK.lock(|| {
+                    crate::debugger::clear_watchpoint(1);
```

Comment:
IMO this should return the previous configuration that we then restore after the write, just in case users end up reconfiguring a different watchpoint. The implementation of these functions also need to be guarded with `LOCK` or they will be racey.